pkgs/development/tools/misc/go-licenses/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
+
+buildGoModule rec {
+  pname = "go-licenses";
+  version = "1.6.0";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "go-licenses";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GAlwTVoVA+n9+EfhybmpKm16FoY9kFzrxy1ZQxS6A8E=";
+  };
+
+  vendorHash = "sha256-ToRn2wj7Yi+UDJwvAhV0ACEhqlcQjt4bRpz7abNRt9A=";
+
+  patches = [
+    # Without this, we get error messages like:
+    # vendor/golang.org/x/sys/unix/syscall.go:83:16: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
+    # The patch was generated by changing "go 1.16" to "go 1.17" and executing `go mod tidy`.
+    ./fix-go-version-error.patch
```

Comment:
Maybe worth upstreaming

Reply:
I'm hesitant to do so because I can't reproduce the issue when building directly with `go build` and I don't know all that much about go packaging.

Reply:
I had a similar issue with another go project recently that only happened when running go vendor in the directory.

Reply:
Thanks for the tip! If I execute `go mod vendor` and then `go build`, I get the error message. I'll try and create a PR upstream.

Reply:
https://github.com/google/go-licenses/pull/192

Reply:
Awesome, thanks!
Do you know why `go mod tidy` changed `go.sum` for you while it did not for me? Maybe this difference is caused by me using macOS?

Reply:
I think I added -compat=1.17 to it.